Sources/ClientRuntime/Networking/Streaming/StreamReader.swift

```diff
@@ -7,14 +7,31 @@ import AwsCommonRuntimeKit
 import class Foundation.FileHandle
 import class Foundation.FileManager
 
+/// Protocol for reading and writing to a stream.
 public protocol StreamReader: AnyObject {
+    /// The number of bytes that are available for reading.
     var availableForRead: UInt { get }
+
+    /// Whether or not the stream has finished writing.
+    /// This is used to determine when to stop reading from the stream.
     var hasFinishedWriting: Bool {get set}
 
     /// Read up to a maximum number of bytes on a stream that is opened..
     /// WARNING:  Be careful as this will read the entire byte stream into memory (up to limit).
+    /// - Parameters: 
+    ///  - maxBytes: The maximum number of bytes to read.  If nil, will read all bytes.
+    /// - rewind: Whether or not to rewind the stream after reading.
     func read(maxBytes: UInt?, rewind: Bool) -> ByteBuffer
-    func seek(offset: Int)
+
+    /// Seek to a specific offset in the stream.
+    /// - Parameter offset: The offset to seek to. This is relative to the beginning of the stream.
+    func seek(offset: Int) throws
+
+    /// Close the stream.
+    /// - Parameter error: The error that caused the stream to close.
     func onError(error: ClientError)
```

Comment:
Nit: Suggest renaming this function to match the intention as defined by the comment. 

```suggestion
    /// Closes the stream when an error occurs.
    /// - Parameter error: The error that caused the stream to close.
    func closeWithError(_ error: ClientError)
```

Reply:
how about just `close`?

Reply:
actually now thinking more, I would like it to be as it is. Because we have exposed this in Input models.

We may do bigger refactor for customer facing structs.

Reply:
Fine to leave it for now since there are a few comments regarding fixing this up in follow-up work.